converters/toZigbee.js

```diff
@@ -838,177 +839,57 @@ const converters = {
-                    command = 'moveToHue';
+            const newColor = libColor.extractColorFromValue(value);
+            const newState = {color: newColor};
+
+            // TODO: define and implement convention what should be stored to state (ex. hsl vs {h,s,v})
+
+            if (newColor.hasOwnProperty('xy')) {
+                value.x = newColor.xy.x;
+                value.y = newColor.xy.y;
+            } else /* HSV */ {
+                const hsv = newColor.hsv;
+                const hsvCorrected = libColor.colorCorrectionsHSV(hsv, meta);
+                if (hsv.hasOwnProperty('value')) {
+                    value.brightness = utils.mapNumberRange(hsvCorrected.value, 0, 100, 0, 254);
                 }
-            } else if (value.hasOwnProperty('s')) {
-                newState.color = {s: value.s};
-                const hsv = utils.gammaCorrectHSV(360, value.s, 100);
-                value.saturation = utils.mapNumberRange(hsv.s, 0, 100, 0, 254);
-                command = 'moveToSaturation';
-            } else if (value.hasOwnProperty('hue') && value.hasOwnProperty('saturation')) {
-                newState.color = {hue: value.hue, saturation: value.saturation};
-                const hsv = utils.gammaCorrectHSV(utils.correctHue(value.hue, meta), value.saturation, 100);
-                value.saturation = utils.mapNumberRange(hsv.s, 0, 100, 0, 254);
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHueAndSaturation';
+                const enhancedHue = utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true);
+                if (enhancedHue) {
+                    value.hue = utils.mapNumberRange(hsvCorrected.hue % 360, 0, 360, 0, 65535);
                 } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHueAndSaturation';
+                    value.hue = utils.mapNumberRange(hsvCorrected.hue, 0, 360, 0, 254);
                 }
-            } else if (value.hasOwnProperty('hue')) {
-                newState.color = {hue: value.hue};
-                const hsv = utils.gammaCorrectHSV(utils.correctHue(value.hue, meta), 100, 100);
-                if (utils.getMetaValue(entity, meta.mapped, 'enhancedHue', 'allEqual', true)) {
-                    value.hue = utils.mapNumberRange(hsv.h % 360, 0, 360, 0, 65535);
-                    command = 'enhancedMoveToHue';
-                } else {
-                    value.hue = utils.mapNumberRange(hsv.h, 0, 360, 0, 254);
-                    command = 'moveToHue';
+                value.saturation = utils.mapNumberRange(hsvCorrected.saturation, 0, 100, 0, 254);
+                if (newColor.hasOwnProperty('hue') && newColor.hasOwnProperty('s')) {
```

Comment:
- newColor -> value
- check for `hue` and `saturation` as we are now using these names in value